src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -923,6 +923,12 @@ def __resolve_dtype__(cls, dtype, data):
         # TODO: These values exist, but I haven't solved them yet
         # binary
         # number
+
+        # Use text dtype for Zarr datasets of strings. Zarr stores variable length strings
+        # as objects, so we need to detect this special case here
+        if hasattr(data, 'attrs') and 'zarr_dtype' in data.attrs and data.attrs['zarr_dtype'] == 'str':
+            return cls.__dtypes['text']
```

Comment:
If we add:
```python
try:
    from zarr import Array as ZarrArray
    import numcodecs
    ZARR_INSTALLED = True
except ImportError:
    ZARR_INSTALLED = False

def _is_zarr_array(value):
    return ZARR_INSTALLED and isinstance(value, ZarrArray)
```

Can we change these lines to:
```python
        if _is_zarr_array(data) and data.filters:
            if numcodecs.VLenUTF8() in data.filters:
                return cls.__dtypes['text']
            elif numcodecs.VLenBytes() in data.filters:
                return cls.__dtypes['ascii']
```

It would be nice if `HDF5IO` can accept any Zarr array just like it accepts any Numpy array, instead of accepting only Zarr arrays that have been written with hdmf-zarr. The Zarr array filters should show that the array dtype is a variable-length string and its encoding.

Reply:
That said, I don't seem to be able to create a test case where these lines (regardless of yours or mine) matters. Could you add one?